azext_iot/common/utility.py

```diff
@@ -146,6 +146,24 @@ def process_json_arg(content: str, argument_name: str = "content", preserve_orde
         )
 
 
+def process_yaml_arg(path: str) -> dict:
+    """Primary processor of yaml file input"""
+
+    if not os.path.exists(path):
+        raise FileOperationError(f"YAML file not found - Please ensure the path '{path}' is correct.")
+
+    try:
+        import yaml
+        with open(path, "rb") as f:
+            return yaml.load(f, Loader=yaml.FullLoader)
```

Comment:
We should use `yaml.safe_load()` or `yaml.SafeLoader` to prevent pickle style attacks / execution of arbitrary code.